### PR TITLE
[Docs] Add config types to all examples

### PR DESCRIPTION
**Pull request by HaNdTriX**
## Summary

- Added jsdoc typing for all examples using `next.config.js`
- Added jsdoc typing for all examples using `tailwind.config.js`

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/active-class-name/next.config.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/** @type {import('next').NextConfig} */
```
Comment (ijjk) [inline examples/active-class-name/next.config.js:1]: I think we need to assign the config to a variable first before exporting for the types to apply correctly, this is why our default template does this

Comment (HaNdTriX) [reply examples/active-class-name/next.config.js:1]: Afaik this issue has been solved. It does not make a difference if you assign to a variable (`const` | `let`) or to an object property (`module = {}`).

Check out the following [CodeSandbox](https://codesandbox.io/s/youthful-archimedes-8czs7i?file=/next.config.js) to test this yourself.

If you use "decorator" functions (`withFoo({})`, `withTranspiledModules({})`, etc.) assigning to a variable makes sense.

@ijjk Please let me now if you still want to rely on this convention. In this case I will adjust the PR.

Comment (HaNdTriX) [reply examples/active-class-name/next.config.js:1]: Check out the tailwind example to see other projects following this "new" convention: https://tailwindcss.com/docs/installation

Comment (ijjk) [reply examples/active-class-name/next.config.js:1]: Ah nice it does look like this is resolved in newer TypeScript versions

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
Thanks!

**Pull request merged**